app/data/views.py

```diff
@@ -561,8 +561,23 @@ def _make_boundary_case(self, boundary_id):
         """
         boundary = Boundary.objects.get(pk=boundary_id)
         polygons = BoundaryPolygon.objects.filter(boundary=boundary)
-        labels = [{'key': str(poly.pk), 'label': poly.data[boundary.display_field]}
-                  for poly in polygons]
+
+        # Sort the polygons by display_field
+        polygons = sorted(polygons, key=lambda p: p.data[boundary.display_field])
+
+        # Remove any items that have an empty label
+        polygons = [p for p in polygons if p.data[boundary.display_field]]
```

Comment:
If you wanted to save a line you could probably move this comprehension up inside the `sorted()` call, but it's fine the way it is, just thought I'd mention it.